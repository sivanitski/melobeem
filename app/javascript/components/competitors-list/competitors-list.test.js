import Enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import React from "react";
import { BrowserRouter as Router } from "react-router-dom";
import renderer from "react-test-renderer";

import CompetitorsList from "./competitors-list";

Enzyme.configure({
  adapter: new Adapter(),
});

const mockProps = {
  competitors: [
    {
      id: `111`,
      name: `Julia Mur`,
      avatar: `some/url/avatar.png`,
      likes: 1000,
      parentName: `John Depp`,
      rank: 2,
    },
    {
      id: `112`,
      name: `Elena Black`,
      avatar: `some/url/avatar.png`,
      likes: 2000,
      parentName: `White Black`,
      rank: 1,
    },
    {
      id: `113`,
      name: `Tom Raddle`,
      avatar: `some/url/avatar.png`,
      likes: 100,
      parentName: `Some Name`,
      rank: 1,
    },
  ],
};

describe("NewIn component", () => {
  it("should render NewIn component", () => {
    const component = renderer
      .create(
        <Router>
          <CompetitorsList competitors={mockProps.competitors} />
        </Router>
      )
      .toJSON();

    expect(component).toMatchSnapshot();
  });
});
