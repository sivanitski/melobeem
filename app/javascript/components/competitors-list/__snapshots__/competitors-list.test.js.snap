// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewIn component should render NewIn component 1`] = `
Array [
  <div
    className="competitors__item"
  >
    <a
      href="/entry/111"
      onClick={[Function]}
    >
      <div
        className="competitors__wrapper"
      >
        <div
          className="competitors__item__img"
        >
          <img
            src="some/url/avatar.png"
          />
        </div>
        <div
          className="competitors__item__names"
        >
          <div
            className="competitors__item__names__child text-black"
          >
            Julia Mur
          </div>
          <div
            className="competitors__item__names__parent text-smaller text-grey"
          >
            John Depp
          </div>
        </div>
        <div
          className="competitors__item__info"
        >
          <div
            className="competitors__item__info__likes"
          >
            <test-file-stub />
            <div
              className="text-smaller text-pink"
            >
              1000
            </div>
          </div>
          <div
            className="competitors__item__info__place text-small text-grey"
          >
            2
          </div>
        </div>
      </div>
    </a>
  </div>,
  <div
    className="competitors__item"
  >
    <a
      href="/entry/112"
      onClick={[Function]}
    >
      <div
        className="competitors__wrapper"
      >
        <div
          className="competitors__item__img"
        >
          <img
            src="some/url/avatar.png"
          />
        </div>
        <div
          className="competitors__item__names"
        >
          <div
            className="competitors__item__names__child text-black"
          >
            Elena Black
          </div>
          <div
            className="competitors__item__names__parent text-smaller text-grey"
          >
            White Black
          </div>
        </div>
        <div
          className="competitors__item__info"
        >
          <div
            className="competitors__item__info__likes"
          >
            <test-file-stub />
            <div
              className="text-smaller text-pink"
            >
              2000
            </div>
          </div>
          <div
            className="competitors__item__info__place text-small text-grey"
          >
            1
          </div>
        </div>
      </div>
    </a>
  </div>,
  <div
    className="competitors__item"
  >
    <a
      href="/entry/113"
      onClick={[Function]}
    >
      <div
        className="competitors__wrapper"
      >
        <div
          className="competitors__item__img"
        >
          <img
            src="some/url/avatar.png"
          />
        </div>
        <div
          className="competitors__item__names"
        >
          <div
            className="competitors__item__names__child text-black"
          >
            Tom Raddle
          </div>
          <div
            className="competitors__item__names__parent text-smaller text-grey"
          >
            Some Name
          </div>
        </div>
        <div
          className="competitors__item__info"
        >
          <div
            className="competitors__item__info__likes"
          >
            <test-file-stub />
            <div
              className="text-smaller text-pink"
            >
              100
            </div>
          </div>
          <div
            className="competitors__item__info__place text-small text-grey"
          >
            1
          </div>
        </div>
      </div>
    </a>
  </div>,
]
`;
